parts/linux/cloud-init/artifacts/aks-check-network.sh

```diff
@@ -81,26 +81,31 @@ function check_and_curl {
     local url=$1
     local error_msg=$2
 
-    # check DNS 
-    nslookup $url > /dev/null
+    # check DNS - cut here to extract domain since acs-mirror health check endpoint is a full URL
+    dnsLookupURL=$(echo $url | cut -d'/' -f1)
+    nslookup $dnsLookupURL > /dev/null
     if [ $? -eq 0 ]; then
-        logs_to_events "AKS.testingTraffic.success" "echo '$(date) - SUCCESS: Successfully tested DNS resolution to $url'"
+        logs_to_events "AKS.testingTraffic.success" "echo '$(date) - SUCCESS: Successfully tested DNS resolution to $dnsLookupURL'"
     else
-        logs_to_events "AKS.testingTraffic.failure" "echo '$(date) - ERROR: Failed to test DNS resolution to $url. $error_msg'"
-        dns_trace $url
+        logs_to_events "AKS.testingTraffic.failure" "echo '$(date) - ERROR: Failed to test DNS resolution to $dnsLookupURL. $error_msg'"
+        dns_trace $dnsLookupURL
         return 1
     fi
 
     local i=0
     while true;
     do
         # curl the url and capture the response code
-        response=$(curl -s -m $MAX_TIME -o /dev/null -w "%{http_code}" "https://${url}" -L)
+        if [ "$url" == "mcr.microsoft.com" ]; then 
+            response=$(curl -s -m $MAX_TIME -o /dev/null -w "%{http_code}" "https://${url}" -L)
+        else 
+            response=$(curl -s -m $MAX_TIME -o /dev/null -w "%{http_code}" "https://${url}" -L --head)
+        fi
 
         if [ $response -ge 200 ] && [ $response -lt 400 ]; then
             logs_to_events "AKS.testingTraffic.success" "echo '$(date) - SUCCESS: Successfully tested $url with returned status code $response'"
             break
-        elif [ $response -eq 400 ] && ([ $url == "acs-mirror.azureedge.net" ] || [ $url == "eastus.data.mcr.microsoft.com" ]); then
+        elif [ $response -eq 400 ] && ([ $url == "eastus.data.mcr.microsoft.com" ]); then
```

Comment:
nit: outside parens are now redundant on the second clause

Reply:
will leave as is for now